app/src/main/java/de/westnordost/streetcomplete/osm/surface/Surface.kt

```diff
@@ -0,0 +1,265 @@
+package de.westnordost.streetcomplete.osm.surface
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.osm.getLastCheckDateKeys
+import de.westnordost.streetcomplete.quests.surface.GENERIC_AREA_SURFACES
+import de.westnordost.streetcomplete.quests.surface.GENERIC_ROAD_SURFACES
+import de.westnordost.streetcomplete.quests.surface.shouldBeDescribed
+
+enum class Surface(val osmValue: String) {
+    ASPHALT("asphalt"),
+    CONCRETE("concrete"),
+    CONCRETE_PLATES("concrete:plates"),
+    CONCRETE_LANES("concrete:lanes"),
+    FINE_GRAVEL("fine_gravel"),
+    PAVING_STONES("paving_stones"),
+    COMPACTED("compacted"),
+    DIRT("dirt"),
+    SETT("sett"),
+    UNHEWN_COBBLESTONE("unhewn_cobblestone"),
+    GRASS_PAVER("grass_paver"),
+    WOOD("wood"),
+    WOODCHIPS("woodchips"),
+    METAL("metal"),
+    GRAVEL("gravel"),
+    PEBBLES("pebblestone"),
+    GRASS("grass"),
+    SAND("sand"),
+    ROCK("rock"),
+    CLAY("clay"),
+    ARTIFICIAL_TURF("artificial_turf"),
+    TARTAN("tartan"),
+    PAVED_ROAD("paved"),
+    UNPAVED_ROAD("unpaved"),
+    GROUND_ROAD("ground"),
+    PAVED_AREA("paved"),
+    UNPAVED_AREA("unpaved"),
+    GROUND_AREA("ground"),
+
+    // extra values, recording duplicates
+    MUD("mud"), // valid value, but not displayed specially in StreetComplete
+    EARTH("earth"), // synonym of "dirt", maybe more clear
+    // this values ideally would be removed from OpenStreetMap, but while they remain
+    // we want to handle them somehow
+    SOIL("soil"),
+    PAVING_STONES_WITH_WEIRD_SUFFIX("paving_stones:30"),
+    COBLLESTONE_FLATTENED("cobblestone:flattened"),
+    BRICK("brick"),
+    BRICKS("bricks");
+
+    companion object {
+        val invalidSurfaces: Set<String> = setOf(
+            "cobblestone", // https://wiki.openstreetmap.org/wiki/Tag%3Asurface%3Dcobblestone
+            "cement", // https://community.openstreetmap.org/t/mysterious-surface-cement/5158 TODO: after 2012-11-23 document that it is an unwanted value at https://wiki.openstreetmap.org/wiki/Tag:surface%3Dconcrete
+        )
+    }
+}
+
+sealed class SurfaceInfo
```

Comment:
(Without looking at whether the construct makes sense, it looks a bit odd to me on first sight:)

SurfaceInfo stuff should rather go into an own file. If the (`create...(): SurfaceInfo` stuff)  is long enough, also rather move that into an own file `SurfaceInfoParser.kt`

Reply:
After some rewrites I am a bit dubious whether SurfaceInfo makes sense